homeassistant/components/generic/diagnostics.py

```diff
@@ -21,11 +21,15 @@
 # A very similar redact function is in components.sql.  Possible to be made common.
 def redact_url(data: str) -> str:
     """Redact credentials from string url."""
-    url = yarl.URL(data)
-    if url.user is not None:
-        url = url.with_user("****")
-    if url.password is not None:
+    url_in = yarl.URL(data)
+    if url_in.user is not None:
+        url = url_in.with_user("****")
+    if url_in.password is not None:
         url = url.with_password("****")
+    if url_in.path is not None:
+        url = url.with_path("****")
+    if url_in.query_string not in [None, ""]:
```

Comment:
```suggestion
    if url_in.query_string:
```

I think you can just check if its truthy here